### ISSUE TITLE
urllib2

### ISSUE THREAD

**Issue by Chris31070** [kvsingh/lyrics-sentiment-analysis#1]
Hi, many thanks for the code, great initiative. I have an issue with the urllib2-module which is not found. I'm working on Python 3.7. I changed the syntax from urllib2 to urllib and the relevant code changes for request, quote and open-methods. Hope it works.

When running the search.py file, I get the following error message:

_Traceback (most recent call last):
  File "C:\Users\Chris\Documents\GitHub\lyrics-sentiment-analysis\search.py", line 100, in <module>
    main()
  File "C:\Users\Chris\Documents\GitHub\lyrics-sentiment-analysis\search.py", line 94, in main
    search_term = arguments[0].translate(None, "\'\"")
IndexError: list index out of range_

I would like to make it the code work for alternative artists and songs. Thanks!

**Comment by Chris31070** [comment]
And a more general question: can you give some guidance of the sequence of running the files? Thanks again!

**Comment by briana-li** [comment]
> Hi, many thanks for the code, great initiative. I have an issue with the urllib2-module which is not found. I'm working on Python 3.7. I changed the syntax from urllib2 to urllib and the relevant code changes for request, quote and open-methods. Hope it works.
> 
> When running the search.py file, I get the following error message:
> 
> _Traceback (most recent call last): File "C:\Users\Chris\Documents\GitHub\lyrics-sentiment-analysis\search.py", line 100, in main() File "C:\Users\Chris\Documents\GitHub\lyrics-sentiment-analysis\search.py", line 94, in main search_term = arguments[0].translate(None, "'"") IndexError: list index out of range_
> 
> I would like to make it the code work for alternative artists and songs. Thanks!

Hi did you figure out how to fix this?

**Comment by Chris31070** [comment]
Working on it using http://python-future.org/compatible_idioms.html (thx Karanveer!)

**Comment by Chris31070** [comment]
Unfortunately still getting errors of different kinds (most recent on the print statement in Song.py line 27). Might this be all due to incompatibility of 2.7 vs 3.7?

**Comment by kvsingh** [comment]
> Unfortunately still getting errors of different kinds (most recent on the print statement in Song.py line 27). Might this be all due to incompatibility of 2.7 vs 3.7?

Yes!
As mentioned in "Essential syntax differences" of the link I sent you, you will need to change 

`print artist, len(urls)`
to 
`print (artist, len(urls))`

For print statements, a bracket needs to be added in python3 (which was not required in python2).

**Comment by Chris31070** [comment]
Thanks again! I indeed changed the syntax, error message is a type error. I'm trying to understand that, see error message below:

RESTART: C:\Users\Chris\Documents\GitHub\lyrics-sentiment-analysis-master\songs.py 
Timeout raised and caught
page 1; num hits 10
Traceback (most recent call last):
  File "C:\Users\Chris\Documents\GitHub\lyrics-sentiment-analysis-master\songs.py", line 27, in <module>
    print(artist, len(urls))
TypeError: object of type 'map' has no len()
>>>_

**Comment by kvsingh** [comment]
Hey Chris31070,

That's another python2 vs python3 issue. You can modify it to `print (artist, len(list(urls))`, or remove the print statement altogether.

**Comment by Chris31070** [comment]
Great, I got it all working now! Next step is to use a long table of artitst and related song. Do you have some tips to do this? The code now uses artist name as search query, I would like to search on a long list of artist-song combination.


Thanks again, it's rather new for me, so any guidance is helpful!

All the best,

Chris

**Comment by Chris31070** [comment]
I got it nearly working, yeahhh. One question: where in the code did you limit the number of results per search to 10? I did not find it in song.py or search.py.

**Comment by kvsingh** [comment]
The genius API, by default, gives 10 results. To get more results, you can use the same query and change the "page" parameter in search.py. Right now, I am returning the results after getting the first page.

**Comment by Chris31070** [comment]
Ah great, thanks. I'm now taking the first hit from the search argument. The search argument is the song title and artist, which gives the right lyrics, so works fine! I would like to add the genre to the data load, saw it is available on the web page of a lyric though not in the api (if I'm not mistaking). So will have to scrape that. 

It's really fun and my first case working through a real application after all tutorials, thank you so much for your code and subsequent support!

**Comment by kvsingh** [comment]
Haha no problem. Glad to help. Would love to see what analysis comes out of this 🙂

**Comment by Chris31070** [comment]
Yes will definitely show it to you, thanks again

**Comment by Chris31070** [comment]
I wrote a short article on https://plakkenenknippen.nl/ The first two graphs are wordclouds of the Dutch Top2000 and its top10. The next two are graphs of the 10 songs with most words and their lexical richness. Next is the same for the top10. The last graph is the sentiment analysis for the top10. I boosted the sentiment by factor 10 to get to this graph. Now I'm scraping the genre to apply the NLTK to that. Fun to do! Many thanks again!
